raiden-ts/tests/e2e/raiden.spec.ts

```diff
@@ -411,55 +412,155 @@ describe('Raiden', () => {
+      // partner's client instance
+      let raiden1: Raiden;
+
+      beforeEach(async () => {
+        raiden1 = await Raiden.create(
+          provider,
+          partner,
+          { ...initialState, address: partner, chainId, registry },
+          info,
+        );
+        await new Promise(resolve => setTimeout(resolve, 1000));
+        await expect(raiden.getAvailability(partner)).resolves.toMatchObject({
+          userId: `@${partner.toLowerCase()}:${matrixServer}`,
+          available: true,
+          ts: expect.any(Number),
+        });
+        await new Promise(resolve => setTimeout(resolve, 100));
+      });
+
+      afterEach(() => raiden1.stop());
+
+      test('success: direct route', async () => {
+        expect.assertions(4);
+
+        const transfers: { [h: string]: RaidenSentTransfer } = {};
+        raiden.transfers$.subscribe(t => (transfers[t.secrethash] = t));
+
+        const secrethash = await raiden.transfer(token, partner, 23);
+        expect(secrethash).toMatch(/^0x[0-9a-fA-F]{64}$/);
+
+        expect(secrethash in transfers).toBe(true);
+        expect(transfers[secrethash].status).toBe(RaidenSentTransferStatus.pending);
+      });
+
+      test('fail: direct channel without enough capacity, pfs disabled', async () => {
+        expect.assertions(2);
+        await expect(raiden.transfer(token, partner, 201)).rejects.toThrowError(
+          /no direct route/i,
+        );
+      });
+
+      test('success: pfs route', async () => {
+        expect.assertions(4);
+
+        const target = accounts[2],
+          raiden2 = await Raiden.create(
+            provider,
+            target,
+            { ...initialState, address: target, chainId, registry },
+            info,
+          );
+
+        await new Promise(resolve => setTimeout(resolve, 1000));
+
+        await raiden1.openChannel(token, target);
+        await raiden1.depositChannel(token, target, 200);
+
+        await expect(raiden.getAvailability(target)).resolves.toMatchObject({
+          userId: `@${target.toLowerCase()}:${matrixServer}`,
+          available: true,
+          ts: expect.any(Number),
+        });
+        await new Promise(resolve => setTimeout(resolve, 100));
+
+        const pfs = 'http://pfs';
+        raiden.config({ pfs });
+        fetch.mockResolvedValueOnce({
+          ok: true,
+          status: 200,
+          json: jest.fn(async () => ({
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            result: [{ path: [partner, target], estimated_fee: 0 }],
+          })),
+          text: jest.fn(async () => ''),
+        });
+
+        await expect(raiden.transfer(token, target, 23)).resolves.toMatch(/^0x[0-9a-fA-F]{64}$/);
+
+        expect(fetch).toHaveBeenCalledWith(
+          expect.stringMatching(new RegExp(`^${pfs}/.*/${tokenNetwork}/paths$`)),
+          expect.objectContaining({ method: 'POST' }),
+        );
+
+        raiden2.stop();
+      });
```

Comment:
I think devs should have the option to get the computed route (e.g. caching it to reuse later, analytics, ...) as this is also something they are going to pay for. 

As you've explained before they can get it from `raiden.transfers$` in `transfer.metadata`. As this is not specifically being mentioned anywhere, it might make sense to include this inside the test case above (not sure, looked most suitable to me)

What do you think?

Reply:
There's comments on the interface itself, but we can improve that, add tests to show it here, and add more documentation somewhere else. I think we have open issues about improving these dev-facing docs.

Reply:
Done, added inspection and comment on test